packages/vite/src/node/ssr/fetchModule.ts

```diff
@@ -77,7 +77,7 @@ export async function fetchModule(
       throw err
     }
     const file = pathToFileURL(resolved.id).toString()
-    const type = isFilePathESM(file, server.config.packageCache)
```

Comment:
When `file` is `file:///C:/Users/green/foo.js`, `file:///C:/Users/green` is passed to `findNearestPackageData`.
https://github.com/vitejs/vite/blob/7440191715b07a50992fcf8c90d07600dffc375e/packages/vite/src/node/utils.ts#L469
Then, `file:\C:\Users\green\package.json` is passed to `tryStatSync`.
https://github.com/vitejs/vite/blob/7440191715b07a50992fcf8c90d07600dffc375e/packages/vite/src/node/packages.ts#L132
`fs.statSync('file:\C:\Users\green\package.json')` throws `Error: ENOENT: no such file or directory, stat 'C:\Users\green\file:C:Usersgreenpackage.json'` error.

I didn't actually create a reproduction though.